compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/Error.hs

```diff
@@ -174,16 +174,25 @@ data UpgradeError
   | RecordFieldsMissing !UpgradedRecordOrigin
   | RecordFieldsExistingChanged !UpgradedRecordOrigin
   | RecordFieldsNewNonOptional !UpgradedRecordOrigin
+  | RecordFieldsOrderChanged !UpgradedRecordOrigin
+  | VariantAddedVariant !UpgradedRecordOrigin
+  | VariantRemovedVariant !UpgradedRecordOrigin
+  | VariantChangedVariantType !UpgradedRecordOrigin
+  | VariantAddedVariantField !UpgradedRecordOrigin
+  | EnumAddedVariant !UpgradedRecordOrigin
+  | EnumRemovedVariant !UpgradedRecordOrigin
```

Comment:
Also, many of these errors are missing additional context about what is missing/added. Was a decision made that these were not required here?

Reply:
https://github.com/digital-asset/daml/issues/18312